### PR TITLE
[MLIR][TORCH] Add E2E support for `prims.convert_element_type` op

### PR DESCRIPTION
**Pull request by vivekkhandelwal1**
Signed-Off By: Vivek Khandelwal<vivek@nod-labs.com>

### CODE REVIEW

File: lib/Dialect/Torch/IR/TorchOps.cpp [hunk 1]
```diff
@@ -730,6 +730,25 @@ OpFoldResult AtenToDtypeOp::fold(ArrayRef<Attribute> operands) {
   return getOperand(0);
 }
 
+//===----------------------------------------------------------------------===//
```
Comment (pashu123) [inline lib/Dialect/Torch/IR/TorchOps.cpp:733]: We don't need this. There is a canonicalize pass after decompose complex ops pass. https://github.com/llvm/torch-mlir/blob/25ab8fcc1f5307258b7f3dbc2a91424e56a0f735/lib/Dialect/Torch/Transforms/Passes.cpp#L136 The canonicalize pass will trigger the canonicalization of `aten.to.dtype` which is already covered.

**Review by pashu123 — CHANGES_REQUESTED** [review]

**Review by pashu123 — APPROVED** [review]
LGTM!

**Comment by silvasean** [comment]
Hey Vivek, can you give more context on where you are seeing all these prims ops from? I wasn't aware that upstream was using those ops yet.

**Comment by vivekkhandelwal1** [comment]
> Hey Vivek, can you give more context on where you are seeing all these prims ops from? I wasn't aware that upstream was using those ops yet.

I'm seeing these ops during the stable diffusion backward graph generation. Even for the DistilGPT2 model, this op has appeared in the IR.

**Pull request merged**